pandas/core/categorical.py

```diff
@@ -374,11 +374,55 @@ def itemsize(self):
 
     def reshape(self, new_shape, *args, **kwargs):
         """
-        An ndarray-compatible method that returns
-        `self` because categorical instances cannot
-        actually be reshaped.
+        DEPRECATED: calling this method will raise an error in a
+        future release.
+
+        An ndarray-compatible method that returns `self` because
+        `Categorical` instances cannot actually be reshaped.
+
+        Parameters
+        ----------
+        new_shape : int or tuple of ints
+            A 1-D array of integers that correspond to the new
+            shape of the `Categorical`. The total size indicated
+            by `new_shape` must be the same as that of the original
+            shape of the `Categorical`. One shape dimension can be
+            negative. In this case, the value is inferred from the
+            length of the array and the remaining dimensions.
         """
+        warn("reshape is deprecated and will raise "
+             "in a subsequent release", FutureWarning, stacklevel=2)
+
         nv.validate_reshape(args, kwargs)
+
+        # while the 'new_shape' parameter has no effect,
+        # we should still enforce valid shape parameters
```

Comment:
That contradicts what you said earlier <a href="https://github.com/pydata/pandas/pull/13012#issuecomment-215135464">here</a>.